uvsim.py

```diff
@@ -76,8 +77,54 @@ def store(self, filename):
             
             print('Program saved successfully')
             
-
-            
+    def convert(self, filename):
+        """
+        Convert the contents of memory to a binary file
+        """
+        op_list = [10, 11, 20, 21, 30, 31, 32, 33, 40, 41, 42, 43]
+        new_list = []
+        file_content = []
+        if filename == "":
+            print("Filename not provided")
+            return
+        try:
```

Comment:
@Kainny2000 This would be a good candidate for refactoring. Writing small unit testable functions would be preferable to nesting so deeply.